plugins/orchestrator-backend/src/helpers/queryBuilder.ts

```diff
@@ -53,8 +59,121 @@ function buildPaginationClause(pagination?: Pagination): string {
   return parts.join(', ');
 }
 
-export function buildFilterCondition(filter?: FilterInfo): string {
-  return filter?.fieldName && filter?.operator && filter?.fieldValue
-    ? `${filter?.fieldName}:{ ${filter?.operator}: ${filter?.fieldValue}}`
-    : '';
+function isLogicalFilter(filter: Filter): filter is LogicalFilter {
+  return (filter as LogicalFilter).filters !== undefined;
+}
+
+export function buildFilterCondition(
+  introspection: IntrospectionField[],
+  filters?: Filter,
+): string {
+  if (!filters) {
+    return '';
+  }
+
+  if (isLogicalFilter(filters)) {
+    if (filters.operator) {
+      const subClauses =
+        filters.filters.map(f => buildFilterCondition(introspection, f)) ?? [];
+      const joinedSubClauses = `${filters.operator.toLowerCase()}: {${subClauses.join(', ')}}`;
+      return joinedSubClauses;
+    }
+  }
+
+  if (!isOperatorSupported(filters.operator)) {
+    throw new Error(`Unsopported operator ${filters.operator}`);
+  }
+
+  let value = filters.value;
+
+  if (filters.operator === FieldFilterOperatorEnum.IsNull) {
+    let booleanValue = false;
+    if (typeof value === 'string') {
+      booleanValue = value.toLowerCase() === 'true';
+    } else if (typeof value === 'number') {
+      booleanValue = value === 1;
+    }
+    return `${filters.field}: {${getGraphQLOperator(filters.operator)}: ${booleanValue}}`;
+  }
+
+  if (Array.isArray(value)) {
+    value = `[${value.map(v => formatValue(filters.field, v, introspection)).join(', ')}]`;
+  } else {
+    value = formatValue(filters.field, value, introspection);
+  }
+
+  if (
```

Comment:
the call to getGraphQLOperator already tests whether the given operator is supported

Reply:
I don't understand this comment, need clarifications.

If you refer to this code block
```js
  if (
    filters.operator === FieldFilterOperatorEnum.Eq ||
    filters.operator === FieldFilterOperatorEnum.Like ||
    filters.operator === FieldFilterOperatorEnum.In
  ) {
    return `${filters.field}: {${getGraphQLOperator(filters.operator)}: ${value}}`;
  }
```
It's necessary to properly build the GraphQL condition, depending on the operator. 
In other words, `eq` `like` and `in` have different syntax then `isNull`.